In [2]:
!pip install empiricaldist # to install

  Created wheel for empiricaldist: filename=empiricaldist-0.6.2-py3-none-any.whl size=10736 sha256=509a0be447ba0f7472d72048ed31544e2e099da9523bdd3b85fa8eae43fc13cf
  Stored in directory: /Users/dominguez/Library/Caches/pip/wheels/2a/ed/75/39cda0596e8f5606df54fb63969c565b49d23869ee64a62435
Successfully built empiricaldist


# Distributions

In [3]:
from empiricaldist import Pmf

# Probability Mass Functions

## PMF
- for dicrete outcomes (ex: head or tails)
- maps each possible outcome to it's probability

In [5]:
# Representing the outcome of a coin toss
coin = Pmf()
coin['heads'] = 1/2
coin['tails'] = 1/2
coin

heads    0.5
tails    0.5
Name: , dtype: float64

In [6]:
# Create a probability mass function for a series of die outcomes
die = Pmf.from_seq([1, 2, 3, 4, 5, 6])
die

1    0.166667
2    0.166667
3    0.166667
4    0.166667
5    0.166667
6    0.166667
Name: , dtype: float64

All outcomes in the sequence appear once, so they all have the same probability, 1/6

In [8]:
letters = Pmf.from_seq(list('Mississippi'))
letters

M    0.090909
i    0.363636
p    0.181818
s    0.363636
Name: , dtype: float64

'M' appears once, so probability = 1/11 = 0.0909

'i' appears 4 times, so probability = 4/11 = 0.3636

and so on...

In [11]:
letters['s']

0.36363636363636365

In [14]:
# Avoid KeyError
try:
    letters['t']
except KeyError as e:
    print("Please choose a letter contained in the Pmf")

Please choose a letter contained in the Pmf


You can also call a Pmf as if it were a function

In [15]:
letters('s')

0.36363636363636365

Calling a quantity that does not exists in the distribution will yield a 0, not an error

In [16]:
letters('t')

0

Can also call some of the elements in the distribution

In [17]:
die([1,4,6])

array([0.16666667, 0.16666667, 0.16666667])

# The Cookie Problem Revisited